src/objects/OrientationGuide.ts

```diff
@@ -0,0 +1,138 @@
+import { Scene } from "phaser";
+import * as CONFIG from "../utils/configuration";
+
+export class OrientationGuide {
+    private scene: Scene;
+    private overlay: Phaser.GameObjects.Container;
+    private backgroundDim: Phaser.GameObjects.Rectangle;
+    private rotateContainer: Phaser.GameObjects.Container;
+    private pulseEffect: Phaser.Tweens.Tween | null = null;
+    
+    constructor(scene: Scene) {
+        this.scene = scene;
+        
+        // Skip if on desktop
+        if (this.isDesktopDevice()) {
+            return;
+        }
+        
+        this.createOverlay();
+        this.checkOrientation();
+    }
+    
+    private isDesktopDevice(): boolean {
+        if (typeof navigator !== 'undefined' && navigator.userAgent) {
```

Comment:
User agent parsing always makes me super nervious, its not reliable and considered kinda bad practice these days.

Can you ask anton or tony for help? I have less experience in this space. Its been years since i did frontend work, so i don't know what is best practices.

Another option would be like the newer [matchMedia](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) apis

A quick test says `matchMedia("(orientation: portrait) and (max-width: 600px)").matches `  (if this is portrait and less than 600px width...i think) works nicely.